src/surf_spot_finder/tracing.py

```
@@ -52,6 +51,15 @@ def setup_tracing(project_name: str, json_tracer: bool) -> TracerProvider:
     else:
         tracer_provider = register(project_name=project_name)
 
-    SmolagentsInstrumentor().instrument(tracer_provider=tracer_provider)
-
     return tracer_provider
+
+
+def setup_tracing(tracer_provider, agent_type):
+    if agent_type == "openai":
+        from openinference.instrumentation.openai import OpenAIInstrumentor
+
+        OpenAIInstrumentor().instrument(tracer_provider=tracer_provider)
+    elif agent_type == "smolagents":
+        from openinference.instrumentation.smolagents import SmolagentsInstrumentor
+
+        SmolagentsInstrumentor().instrument(tracer_provider=tracer_provider)
```

Can we make this an enum, and/or throw an error if agent_type is invalid? Function signature is missing typing and description